src/services/posts/postsService.js

```diff
@@ -1,7 +1,94 @@
 import { createAsyncThunk } from "@reduxjs/toolkit";
-import { useApi } from "../../hooks/useApi";
+import { callApi } from "../../utils";
 export const fetchPosts = createAsyncThunk("posts/fetchPosts", async () => {
-    const { callApi } = useApi();
   const response = await callApi("get", "posts");
   return response.data;
 });
+export const fetchBookmarkedPosts = createAsyncThunk(
+  "posts/fetchBookmarkedPosts",
+  async () => {
+    const response = await callApi("get", "user/posts/bookmarked", true);
+    return response.data;
+  }
+);
+export const fetchDraftPosts = createAsyncThunk(
+  "posts/fetchDraftPosts",
+  async () => {
+    const response = await callApi("get", "user/posts/draft", true);
```

Comment:
Couldn't find any try catch block in network calls. Since, you're using thunk, I believe you're relying on it's internal try catch handler